src/interface/blockchain.cpp

```diff
@@ -172,6 +173,47 @@ void blockchain::last_height_fetched(const code& ec, size_t last_height,
     handler(message(request, std::move(result)));
 }
 
+void blockchain::fetch_block(server_node& node, const message& request,
+    send_handler handler)
+{
+    const auto& data = request.data();
+
+    if (data.size() == hash_size)
+        blockchain::fetch_block_by_hash(node, request, handler);
+    else if (data.size() == sizeof(uint32_t))
+        blockchain::fetch_block_by_height(node, request, handler);
+    else
+        handler(message(request, error::bad_stream));
+}
+
+void blockchain::fetch_block_by_hash(server_node& node,
+    const message& request, send_handler handler)
+{
+    const auto& data = request.data();
+    BITCOIN_ASSERT(data.size() == hash_size);
+
+    auto deserial = make_safe_deserializer(data.begin(), data.end());
+    const auto block_hash = deserial.read_hash();
+
+    node.chain().fetch_block(block_hash, node.blockchain_settings().bip141,
```

Comment:
@evoskuil Is this a sane way to pass the `witness` bool?

Reply:
This is probably the better way to go, rather than checking the config setting directly:

```
    const auto witness = script::is_enabled(
        node.blockchain_settings().enabled_forks(), rule_fork::bip141_rule);

    node.chain().fetch_block(block_hash, witness,
        std::bind(&blockchain::block_fetched,
            _1, _2, request, handler));
```

Reply:
Noting here that since it's a new API for v4, we could just hardcode true for this parameter, similar to fetch_transaction2.  Alternatively, I could update fetch_transaction2 to use this style so that if the node isn't configured for it, it won't use it (similar to fetch_transaction, but configurable).

I'll probably opt for the latter and we can re-visit and re-organize as needed.